packages/tooltip/src/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,155 @@
+  forceMount,
+  side = "bottom",
+  sideOffset = 4,
+  align = "center",
+  alignOffset,
+  avoidCollisions,
+  collisionBoundary,
+  collisionPadding,
+  arrowPadding,
+  sticky,
+  hideWhenDetached,
+  // Box Props
+  label,
+  labelProps,
+  children,
+}: TooltipProps<T>) => {
+  const deriveAnimationBasedOnSide = (side: TooltipProps<T>["side"]) => {
+    const ANIMATION_OFFSET = 5;
+    if (side === "top") {
+      return {
+        y: -ANIMATION_OFFSET,
+      };
+    }
+
+    if (side === "bottom") {
+      return {
+        y: ANIMATION_OFFSET,
+      };
+    }
+
+    if (side === "left") {
+      return {
+        x: -ANIMATION_OFFSET,
+      };
+    }
+
+    if (side === "right") {
+      return {
+        x: ANIMATION_OFFSET,
+      };
+    }
+  };
+  return (
+    <RadixTooltip.Provider
+      delayDuration={delayDuration}
+      skipDelayDuration={skipDelayDuration}
+      disableHoverableContent={disableHoverableContent}
+    >
+      <RadixTooltip.Root
+        defaultOpen={defaultOpen}
+        open={open}
+        onOpenChange={onOpenChange}
+      >
+        <RadixTooltip.Trigger asChild={true}>
+          <RefToTgphRef>{children}</RefToTgphRef>
+        </RadixTooltip.Trigger>
+        <RadixTooltip.Portal>
+          <InvertedAppearance asChild>
+            <RadixTooltip.Content
+              aria-label={ariaLabel}
+              onEscapeKeyDown={onEscapeKeyDown}
+              onPointerDownOutside={onPointerDownOutside}
+              forceMount={forceMount}
+              side={side}
+              sideOffset={sideOffset}
+              align={align}
+              alignOffset={alignOffset}
+              avoidCollisions={avoidCollisions}
+              collisionBoundary={collisionBoundary}
+              collisionPadding={collisionPadding}
+              arrowPadding={arrowPadding}
+              sticky={sticky}
+              hideWhenDetached={hideWhenDetached}
+            >
+              <Box
+                as={motion.div}
+                // Add tgph class so that this always works in portals
+                className="tgph"
+                initial={{
+                  opacity: 0.5,
+                  scale: 0.6,
+                  ...deriveAnimationBasedOnSide(side),
+                }}
+                animate={{
+                  opacity: 1,
+                  scale: 1,
+                  x: 0,
+                  y: 0,
+                }}
+                transition={{ duration: 0.2, type: "spring", bounce: 0 }}
+                bg="gray-1"
+                rounded="3"
+                py="2"
+                px="3"
+                style={{
+                  transformOrigin:
+                    "var(--radix-tooltip-content-transform-origin)",
+                }}
+                {...(labelProps ? { labelProps } : {})}
+              >
+                <RadixTooltip.Arrow fill="var(--tgph-gray-1)" />
+                {typeof label === "string" ? (
+                  <Text as="span" size="1">
```

Comment:
In Storybook, the text looks a bit off center vertically. Haven't looked too closely at it, maybe something with the line height or the tooltip could be a stack and center everything.

![image](https://github.com/knocklabs/telegraph/assets/12838032/0f7f880a-d276-4d6c-864f-16c56f8a3ad3)